### PR TITLE
Use Node extensions instead of *splat

### PR DESCRIPTION
**Pull request by bquorning**
Use more rubocop-ast and less `*` splatting.

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [ ] Added tests.
- [ ] Updated documentation.
- [ ] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Review by pirj — APPROVED** [review]
Thank you!

**Comment by pirj** [comment]
With such a coverage now it's almost a blind approve 😄

**Comment by bquorning** [comment]
@ydah It looks like mdformat checks are broken, probably related to https://github.com/executablebooks/mdformat/issues/480. Do you want to have a look at it?

File: lib/rubocop/cop/rspec/single_argument_message_chain.rb [hunk 1]
```diff
@@ -67,11 +67,9 @@ def single_element_array?(node)
         end
 
         def autocorrect_hash_arg(corrector, arg)
-          key, value = *arg.children.first
-
-          corrector.replace(arg, key_to_arg(key))
+          corrector.replace(arg, key_to_arg(arg.keys.first))
```
Comment (Darhazer) [inline lib/rubocop/cop/rspec/single_argument_message_chain.rb:70]: how about
```ruby
pair = arg.pairs.first
```
and then using `key_to_arg(pair.key)` and `pair.value.source` respectively ?

**Review by Darhazer — COMMENTED** [review]
Always good to see more node specific methods usage and less deconstruction

**Review by Darhazer — APPROVED** [review]

File: lib/rubocop/cop/rspec/single_argument_message_chain.rb (hunk 1 above)
Comment (bquorning) [reply lib/rubocop/cop/rspec/single_argument_message_chain.rb]: Yep, that’s even better. Thank you.

**Pull request merged**